In [23]:
#imports
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt 
import statsmodels.api as sm



Build a regression model using Python’s statsmodels module that demonstrates a relationship between the number of bikes in a particular location and the characteristics of the POIs in that location.

In [28]:
#first import relevant Dfs from previous steps 
ratings_data = pd.read_csv('yelpdf.csv')
citybikes_data = pd.read_csv('clean_citybikesdata.csv')
#test

citybikes_data.head()
ratings_data.head()


name                                           location  rating
0         Bon Temps  {'address1': '61 Young Street', 'address2': No...     5.0
1             Tomah  {'address1': '132 Queen Street S', 'address2':...     5.0
2     Famous Recipe  {'address1': '158 King Street W', 'address2': ...     5.0
3  The Burnt Tongue  {'address1': '182 Locke Street S', 'address2':...     4.5
4    Ark and Anchor  {'address1': '300 King Street W', 'address2': ...     4.5

In [29]:
#define independant and dependant variables 
X = citybikes_data['free_bikes']
y = ratings_data['rating']


#add intercept to the model 
X = sm.add_constant(X)
#create model to fit linear regression model 
Linreg_model = sm.OLS(y, X).fit()



In [30]:
#print model summary and analyze results 
print(Linreg_model.summary())

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.034
Model:                            OLS   Adj. R-squared:                  0.014
Method:                 Least Squares   F-statistic:                     1.697
Date:                Sat, 28 Oct 2023   Prob (F-statistic):              0.199
Time:                        00:39:04   Log-Likelihood:                -39.846
No. Observations:                  50   AIC:                             83.69
Df Residuals:                      48   BIC:                             87.52
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          3.8736      0.149     25.923      0.0

Interpret results. Expand on the model output, and derive insights from your model.


As we saw from the scatter plot I used earlier in the previous step (shown in picture for ref (table1)) there is little to no correlation between restaurant rating and number of bikes at any given bike station in Hamilton. 
The adj R squared of 0.014 indicates to me a negligable positive correlation and basically is saying that its random whether a restaurant has a high rating and a high number of bikes or vice versa etc 

Table 1 
![Alt text](image.png)

After some reflection I learned that the API data I had pulled was in fact incomplete and I ran statitical analysis when I knew the data would be limited. I tried my best to keep the parameters clean and simple but I expected my results to not be as meaningful.
If I had more time, I would have chose a city with a more complete API data set and starting with a hypothesis on a different POI and its potential relationship to citybikes. 
